CA panic when customer sets a wrong VMSS name

CA would panic when customer starts it with a wrong VMSS name. We should log an error instead of panic.

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x185c3d5]


 

goroutine 56 [running]:

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.(*ScaleSet).getCurSize(0xc000556bd0, 0x0, 0x0, 0x0)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go:196 +0x405

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.(*ScaleSet).Nodes(0xc000556bd0, 0x0, 0x0, 0x0, 0x0, 0x0)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go:567 +0x62

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.(*asgCache).regenerate(0xc0009b4040, 0x2d2d9a8, 0xc0009b4068)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_cache.go:177 +0x83

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.(*AzureManager).regenerateCache(0xc0009844e0, 0x0, 0x0)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_manager.go:330 +0x8c

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.(*AzureManager).fetchExplicitAsgs(0xc0009844e0, 0xc0001cd6a0, 0x2, 0x2, 0x0, 0x0)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_manager.go:215 +0x18b

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.CreateAzureManager(0x0, 0x0, 0xc0001cd6a0, 0x2, 0x2, 0x0, 0x0, 0x0, 0xc00066cd78, 0x0, ...)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_manager.go:190 +0x45c

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure.BuildAzure(0xa, 0x3fe0000000000000, 0x8bb2c97000, 0x1176592e000, 0x0, 0x4e200, 0x0, 0x186a0000000000, 0x0, 0x4f2b780, ...)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_cloud_provider.go:141 +0x1b5

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/builder.buildCloudProvider(0xa, 0x3fe0000000000000, 0x8bb2c97000, 0x1176592e000, 0x0, 0x4e200, 0x0, 0x186a0000000000, 0x0, 0x4f2b780, ...)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/builder/builder_all.go:54 +0x4b4

k8s.io/autoscaler/cluster-autoscaler/cloudprovider/builder.NewCloudProvider(0xa, 0x3fe0000000000000, 0x8bb2c97000, 0x1176592e000, 0x0, 0x4e200, 0x0, 0x186a0000000000, 0x0, 0x4f2b780, ...)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/builder/cloud_provider_builder.go:45 +0x1cc

k8s.io/autoscaler/cluster-autoscaler/core.initializeDefaultOptions(0xc001023888, 0xc0005c27e0, 0x14)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/core/autoscaler.go:93 +0x309

k8s.io/autoscaler/cluster-autoscaler/core.NewAutoscaler(0xa, 0x3fe0000000000000, 0x8bb2c97000, 0x1176592e000, 0x0, 0x4e200, 0x0, 0x186a0000000000, 0x0, 0x4f2b780, ...)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/core/autoscaler.go:62 +0x43

main.buildAutoscaler(0xc000086000, 0x0, 0xc00033a758, 0xc0003c2f68)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/main.go:298 +0x37f

main.run(0xc0005276d0)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/main.go:304 +0x39

main.main.func2(0x3171600, 0xc00033ac00)

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/main.go:396 +0x2a

created by k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run

 /gopath/src/k8s.io/autoscaler/cluster-autoscaler/vendor/k8s.io/client-go/tools/leaderelection/leaderelection.go:182 +0xec

rpc error: code = Unknown desc = Error: No such container: c8aee4f7ece2999150435f1929fb4c0d697db98f8161b8749f44341a1aec20dc% 
```

Related code:

https://github.com/kubernetes/autoscaler/blob/17536d5bb0037842a77c71b7d7784b7c39603ade/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go#L196

/kind bug
/area provider/azure

--- a/cluster-autoscaler/cloudprovider/azure/azure_cache.go
+++ b/cluster-autoscaler/cloudprovider/azure/azure_cache.go
@@ -174,7 +174,7 @@ func (m *asgCache) FindForInstance(instance *azureRef, vmType string) (cloudprov
 		return asg, nil
 	}
 
-	klog.Warningf("FindForInstance: Couldn't find NodeGroup of instance %q", inst)
+	klog.V(4).Infof("FindForInstance: Couldn't find NodeGroup of instance %q", inst)
 	return nil, nil
 }
 

--- a/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go
+++ b/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go
@@ -157,6 +157,10 @@ func (scaleSet *ScaleSet) getVMSSInfo() (compute.VirtualMachineScaleSet, error) 
 	scaleSetStatusCache.lastRefresh = time.Now()
 	scaleSetStatusCache.scaleSets = newStatus
 
+	if _, exists := scaleSetStatusCache.scaleSets[scaleSet.Name]; !exists {
+		return compute.VirtualMachineScaleSet{}, fmt.Errorf("could not find vmss: %s", scaleSet.Name)
+	}
+
 	return scaleSetStatusCache.scaleSets[scaleSet.Name], nil
 }
 

